pygfx/materials/_base.py

```diff
@@ -39,15 +39,15 @@ def _set_size_of_uniform_array(self, key, new_length):
             return  # early exit
 
         dtype = self.uniform_type[key]
-        shape = dtype[1]
-        assert len(dtype) == 2
-        assert len(shape) == 3, f"uniform field {key} does not look like an array"
-        # This is true unless someone fooled with the data: current_length == shape[0]
-        # And it will certainly be true when we're done.
+        assert (
+            "*" in dtype
+        ), f"uniform field {key} '{dtype}' does not look like an array"
+        n, _, subtype = dtype.partition("*")
+        assert int(n) >= 0
 
         # Adjust type definition (note that this is originally a class attr)
         self.uniform_type = self.uniform_type.copy()
-        self.uniform_type[key] = dtype[0], (new_length, shape[1], shape[2])
+        self.uniform_type[key] = f"{new_length}*" + subtype
```

Comment:
Why concatenate instead of including it in the f-string?

Reply:
Good point